src/arvgvfakecamera.c

```diff
@@ -483,20 +500,46 @@ arv_gv_fake_camera_start (ArvGvFakeCamera *gv_fake_camera)
 			}
 			g_clear_object (&inet_socket_address);
 			g_clear_object (&inet_address);
-
-			g_free (gvcp_address_string);
+                        
+                        /* subnet broadcast discovery */
+			socket_address = g_socket_address_new_from_native (ifap_iter->ifa_broadaddr,
+									   sizeof (struct sockaddr));
+                        inet_address = g_inet_socket_address_get_address (G_INET_SOCKET_ADDRESS (socket_address));
+			discovery_address_string = g_inet_address_to_string (inet_address);
+			arv_debug_device ("[GvFakeCamera::start] Discovery address = %s", discovery_address_string);
 			g_free (discovery_address_string);
+			inet_socket_address = g_inet_socket_address_new (inet_address, ARV_GVCP_PORT);
+			if (g_strcmp0 (gvcp_address_string, discovery_address_string) == 0)
+				gv_fake_camera->priv->subnet_discovery_socket = NULL;
+			else {
+				gv_fake_camera->priv->subnet_discovery_socket = g_socket_new (G_SOCKET_FAMILY_IPV4,
+									    G_SOCKET_TYPE_DATAGRAM,
+									    G_SOCKET_PROTOCOL_UDP, NULL);
+				if (!g_socket_bind (gv_fake_camera->priv->subnet_discovery_socket, inet_socket_address, FALSE, NULL))
+					binding_error = TRUE;
+				g_socket_set_blocking (gv_fake_camera->priv->subnet_discovery_socket, FALSE);
+			}
+			g_clear_object (&socket_address);
+			g_clear_object (&inet_socket_address);
+			g_clear_object (&inet_address);
+                        
 
 			gv_fake_camera->priv->gvcp_fds[0].fd = g_socket_get_fd (gv_fake_camera->priv->gvcp_socket);
 			gv_fake_camera->priv->gvcp_fds[0].events = G_IO_IN;
 			gv_fake_camera->priv->gvcp_fds[0].revents = 0;
+                        gv_fake_camera->priv->n_gvcp_fds = 1;
 			if (gv_fake_camera->priv->discovery_socket != NULL) {
 				gv_fake_camera->priv->gvcp_fds[1].fd = g_socket_get_fd (gv_fake_camera->priv->discovery_socket);
 				gv_fake_camera->priv->gvcp_fds[1].events = G_IO_IN;
 				gv_fake_camera->priv->gvcp_fds[1].revents = 0;
-				gv_fake_camera->priv->n_gvcp_fds = 2;
-			} else
-				gv_fake_camera->priv->n_gvcp_fds = 1;
+				gv_fake_camera->priv->n_gvcp_fds++;
+                                if (gv_fake_camera->priv->subnet_discovery_socket != NULL) {
+				    gv_fake_camera->priv->gvcp_fds[2].fd = g_socket_get_fd (gv_fake_camera->priv->subnet_discovery_socket);
+				    gv_fake_camera->priv->gvcp_fds[2].events = G_IO_IN;
+				    gv_fake_camera->priv->gvcp_fds[2].revents = 0;
+				    gv_fake_camera->priv->n_gvcp_fds++;
+                                }
+			} 
```

Comment:
Trailing whitespace

Reply:
Can't see it

Reply:
![Screenshot_2019-08-12 Feature subnet discovery port by highperformancecoder · Pull Request #271 · AravisProject aravis](https://user-images.githubusercontent.com/136864/62874222-18411680-bd21-11e9-877b-66080a544fd0.png)